src/main/java/com/venafi/vcert/sdk/connectors/cloud/CloudConnectorUtils.java

```diff
@@ -179,27 +173,106 @@ private static void addCitToApp(CertificateIssuingTemplate cit, Application appl
         }
 
-        //if the App doesn't contains the relation to the cit
+        //if the App doesn't contain the relation to the cit
         if ( !citAliasIdMap.containsKey(cit.name()) ) {
             //adding the reference to the cit
             citAliasIdMap.put(cit.name(), cit.id());
 
-            //getting the appId because it will used to invoke the API to update the related Application
+            //getting the appId because it will be used to invoke the API to update the related Application
             String appId = application.id();
 
-            //The id, companyId, fqDns and internalFqDns needs to be null in the request to update the Application
-            // so therefore these attributes are set to null
+            //The id, companyId, fqDns and internalFqDns needs to be null in the request to update the Application,
+            //therefore these attributes are set to null
             application.id(null);
             application.companyId(null);
             application.fqDns(null);
             application.internalFqDns(null);
 
+			// Updating the owners list
+			List<Application.OwnerIdsAndType> ownersList =  CloudConnectorUtils.resolveOwners(usersList, apiKey, cloud);
+			application.ownerIdsAndTypes(ownersList);
+
             cloud.updateApplication(application, appId, apiKey);
         }
     }
 
+	private static List<Application.OwnerIdsAndType> resolveOwners(String[] usersList, String apiKey, Cloud cloud) {
+		List<Application.OwnerIdsAndType> ownersList = new ArrayList<>();
+
+		if (usersList == null) {
+			// When no user is provided on the list, adds the current one as owner
+			UserDetails userDetails = cloud.authorize(apiKey);
+			String userId = userDetails.user().id();
+			Application.OwnerIdsAndType currentOwner = new Application.OwnerIdsAndType();
+			currentOwner.ownerId(userId);
+			currentOwner.ownerType(CloudConstants.OWNER_TYPE_USER);
+			ownersList.add(currentOwner);
+		}
+		else {
+			// Resolving the usernames list
+			// Creating a higher level Teams object to cache the response.
+			Teams tResponse = null;
+			for (String username: usersList) {
+				UserResponse response = cloud.retrieveUser(username, apiKey);
+				// If the name matches a user, create the entry
+				if (response != null) {
+					Application.OwnerIdsAndType owner = new Application.OwnerIdsAndType();
+					owner.ownerId(response.users().get(0).id());
+					owner.ownerType(CloudConstants.OWNER_TYPE_USER);
+					ownersList.add(owner);
+				}else{
+					if (tResponse == null) {
+						tResponse = cloud.retrieveTeams(apiKey);
+					}
+					if (tResponse != null) {
+						for (Team t : tResponse.teams()) {
+							if (t.name().equals(username)) {
+								Application.OwnerIdsAndType owner = new Application.OwnerIdsAndType();
+								owner.ownerId(t.id());
+								owner.ownerType(CloudConstants.OWNER_TYPE_TEAM);
+								ownersList.add(owner);
+								break;
+							}
+						}
+					}
+
+				}
```

Comment:
Please extract it in a new method

Reply:
The method highlighted here is the actual one that resolves the friendly names to the owners objects from VaaS. I dont think is necessary to move this piece of logic to another method.